src/type.rs

```diff
@@ -285,9 +285,11 @@ impl Printable for TypeObj {
         &self,
         ctx: &Context,
         state: &printable::State,
-        f: &mut core::fmt::Formatter<'_>,
-    ) -> core::fmt::Result {
-        Printable::fmt(self.as_ref(), ctx, state, f)
+        f: &mut std::fmt::Formatter<'_>,
+    ) -> std::fmt::Result {
+        self.get_type_id().fmt(ctx, state, f)?;
+        write!(f, " ")?;
+        <dyn Type as Printable>::fmt(self.deref(), ctx, state, f)
```

Comment:
Why does this need `<dyn Type as Printable>` but `impl Printable for AttrObj` doesn't need it?

Reply:
because `Type` requires `Printable` and `Debug` and here the compiler is complaining about multiple applicable items for `fmt`:

```
error[E0034]: multiple applicable items in scope
   --> src/type.rs:293:22
    |
293 |         self.deref().fmt(ctx, state, f)
    |                      ^^^ multiple `fmt` found
    |
note: candidate #1 is defined in the trait `printable::Printable`
   --> src/printable.rs:129:5
    |
129 |     fn fmt(&self, ctx: &Context, state: &State, f: &mut fmt::Formatter<'_>) -> fmt::Result;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: candidate #2 is defined in the trait `Debug`
   --> /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/fmt/mod.rs:603:5
help: disambiguate the method for candidate #1
    |
293 |         printable::Printable::fmt(&self.deref(), ctx, state, f)
    |
help: disambiguate the method for candidate #2
    |
293 |         Debug::fmt(&self.deref(), ctx, state, f)
    |
```